packages/flutter/lib/src/widgets/scroll_notification.dart

```diff
@@ -24,20 +24,29 @@ mixin ViewportNotificationMixin on Notification {
   int get depth => _depth;
   int _depth = 0;
 
-  @override
-  bool visitAncestor(Element element) {
-    if (element is RenderObjectElement && element.renderObject is RenderAbstractViewport)
-      _depth += 1;
-    return super.visitAncestor(element);
-  }
-
   @override
   void debugFillDescription(List<String> description) {
     super.debugFillDescription(description);
     description.add('depth: $depth (${ depth == 0 ? "local" : "remote"})');
   }
 }
 
+/// A mixin that allows [Element]s containing Viewport like widgets to correctly
+/// modify the notification depth of a [ViewportNotificationMixin].
+///
+/// See also:
+///   * [Viewport], which creates a custom [MultiChildRenderObjectElement] that mixes
+///     this in.
+mixin ViewportElement on NotifiableElementMixin {
```

Comment:
We should probably link to this from some other docs as well so people implementing custom viewports find it.

Reply:
What is a good place where we describe this - on Viewport itself?

Reply:
added a See More section